src/python/grpcio_observability/make_grpcio_observability.py

```diff
@@ -143,10 +162,28 @@ def _generate_deps_file_content():
     # Collect .cc files (that will be later included in the native extension build)
     cc_files = set()
     for name in cc_files_output:
+        if name.endswith(".proto"):
+            filepath = _bazel_proto_name_to_file_path(name)
+            if filepath and os.path.exists(filepath.replace("grpc_root/", "")):
+                cc_files.add(filepath)
         if name.endswith(".cc"):
             filepath = _bazel_name_to_file_path(name)
             if filepath:
                 cc_files.add(filepath)
+        if (
+            name.endswith(".c")
+            and name.startswith("@com_google_protobuf//upb/")
+            and ("stage0" not in name)
+            and ("stage1" not in name)
+            and ("upb/text" not in name)
+            and ("upb/wire:internal/decode_fast.c" not in name)
+            and ("upb/wire:decode.c" not in name)
+        ):
```

Comment:
upb/text, upb/wire* should be removed here

then we end up with a compile error trying to include utf8_ranges.h